### PR TITLE
Coordinate system

### PR DESCRIPTION
**Pull request by nic-schi**
### Coordinate System
A Coordinate-System can now be created using an React-Component. This was one of the required functionalities for the portfolio analysis. The Coordinate-System then looks like the following:

image

By hovering over an Point in the System an Tooltip will open up and the X- and Y-Values will be displayed.

The X And Y Values are written in an table under the Coordinate System.

image

### CODE REVIEW

File: src/general-components/CoordinateSystem/ValueRenderer/ValueRenderer.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+abstract class ValueRenderer {
```
Comment (Geist5000) [inline src/general-components/CoordinateSystem/ValueRenderer/ValueRenderer.tsx:1]: I think this should be an interface

**Review by Geist5000 — CHANGES_REQUESTED** [review]
There is documentation missing. In my oppinion at least the props should get documented

File: src/general-components/CoordinateSystem/CoordinateSystem.tsx [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+                                            height: String(size) + "%"
+                                        }}
+                                    >
+                                        <OverlayTrigger
+                                            trigger={["hover", "focus"]}
+                                            placement="top"
+                                            overlay={(
+                                                <Tooltip>
+                                                    {(this.props.tooltipContentRenderer) ? this.props.tooltipContentRenderer(point) : ""}
+                                                </Tooltip>
```
[the remark this answers is not in the document]
Comment (Geist5000) [reply src/general-components/CoordinateSystem/CoordinateSystem.tsx:141]: This is just an information.

[the remark this answers is not in the document]
Comment (ma1160) [reply src/general-components/CoordinateSystem/CoordinateSystem.tsx:141]: Will be looked at a later date.

**Review by Geist5000 — APPROVED** [review]
Looks good to me!

**Pull request merged**